tests/unit/conftest.py

```diff
@@ -444,14 +446,33 @@ def lifecycle_service(default_project, default_factory, default_build_plan, tmp_
 
 
 @pytest.fixture()
-def package_service(default_project, default_factory):
+def provider_service(default_project, default_factory, default_build_plan, tmp_path):
+    from snapcraft.application import APP_METADATA
+    from snapcraft.services import Provider as ProviderSvc
+
+    return ProviderSvc(
+        app=APP_METADATA,
+        services=default_factory,
+        project=default_project,
+        work_dir=tmp_path / "work",
+        build_plan=default_build_plan,
+        install_snap=False,
+    )
+
+
+@pytest.fixture()
+def package_service(default_project, default_factory, snapcraft_yaml, tmp_path):
     from snapcraft.application import APP_METADATA
     from snapcraft.services import Package
 
+    file_path = tmp_path / "snapcraft.yaml"
+    snapcraft_yaml(filename=file_path)
+
     return Package(
         app=APP_METADATA,
         project=default_project,
         services=default_factory,
+        project_path=file_path,
```

Comment:
the project is much more that the snapcraft.yaml, we also have hooks, gui and other assets, I would consider project_path the snapcraft project root directory

Reply:
in the case of charms, the project_path can also include metadata.yaml, action.yaml. It seems a bit wasteful to only use it for one yaml

Reply:
I suggest we refactor this when adding hooks and assets to the payload so we can limit the scope of this PR and have it landed.